hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -466,15 +466,15 @@ public IndexFilter getFilterPredicates(Configuration configuration) {
    * get data blocks of given segment
    */
   protected List<CarbonInputSplit> getDataBlocksOfSegment(JobContext job, CarbonTable carbonTable,
-      IndexFilter expression, List<Segment> segmentIds,
+      IndexFilter expression, List<Segment> validSegments,
```

Comment:
revert this change.. Not related to this PR

Reply:
Actually when the main table was giving 0 rows. I was debugging this flow. And this naming is wrong. segmentID should be string, here it is not the segmentID but the complete Segment object. so renamed. Still you want to revert it ?

Reply:
its fine.. ill merge